lib/topic_query.rb

```diff
@@ -869,7 +869,34 @@ def remove_muted_topics(list, user)
   def remove_muted_categories(list, user, opts = nil)
     category_id = get_category_id(opts[:exclude]) if opts
 
-    if user
+    if SiteSetting.mute_all_categories_by_default
+      if user
+        list = list.references("cu")
+          .where("
+          NOT EXISTS (
+            SELECT 1
+              FROM categories c
+              LEFT OUTER JOIN category_users cu
+              ON c.id = cu.category_id AND cu.user_id = :user_id
+            WHERE c.id = topics.category_id
+              AND c.id <> :category_id
+              AND (cu.notification_level IS NULL OR cu.notification_level = :muted)
```

Comment:
Small tip here. You can use `COALESCE`:

`AND (COALESCE(cu.notification_level, :muted) = :muted)`

(same for the next line.)